notify-slack

```diff
@@ -31,17 +31,37 @@ text="${1:?Some message text is required.}"
 
 if [[ "$upload" == 1 ]]; then
     echo "Uploading data to Slack with the message: $text"
-    curl https://slack.com/api/files.upload \
+
+    upload_file="$(mktemp -t upload-file-XXXXXX)"
+    trap "rm -f '$upload_file'" EXIT
+
+    cat /dev/stdin > "$upload_file"
+    length=$(printf '%d' "$(<"$upload_file" wc -c)")
+
+    upload_info=$(curl https://slack.com/api/files.getUploadURLExternal \
         --header "Authorization: Bearer $SLACK_TOKEN" \
-        --form-string channels="$SLACK_CHANNELS" \
-        --form-string title="$text" \
-        --form-string filename="$text" \
+        --form filename="$text" \
+        --form length="$length" \
```

Comment:
Keep use of `--form-string` to avoid curl from interpreting initial metachars (e.g. `@` or `<`) in values we pass.

```suggestion
        --form-string filename="$text" \
        --form-string length="$length" \
```